src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -878,7 +878,7 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
             else
             {
                 // If there are any non-event fields, they are at the very beginning.
-                IEnumerable<FieldSymbol> nonEventFields = GetMembers<FieldSymbol>(this.GetMembers(), SymbolKind.Field, offset: 0).Select(f => f.TupleUnderlyingField ?? f);
+                IEnumerable<FieldSymbol> nonEventFields = GetMembers<FieldSymbol>(this.GetMembers(), SymbolKind.Field, offset: 0);
```

Comment:
>GetMembers<FieldSymbol>(this.GetMembers(), SymbolKind.Field, offset: 0); [](start = 58, length = 72)

Is it possible to run into a ```TupleErrorFieldSymbol``` here? We don't want it in the result.

Reply:
It turns out it's not possible, but the reason seems suspicious. The tuple error fields are not at the very begining... They come after the constructor, so they are filtered out by `GetMembers<FieldSymbol>(,,, SymbolKind.Field, ...)`.
I'll put a filter on `this.GetMembers()` to strengthen regardless.

---
In reply to: [426204181](https://github.com/dotnet/roslyn/pull/44231#discussion_r426204181) [](ancestors = 426204181)

Reply:
> The tuple error fields are not at the very begining... They come after the constructor, so they are filtered out by GetMembers<FieldSymbol>(,,, SymbolKind.Field, ...).

This doesn't sound right. I think the members should be grouped as we expect and that consumers of ```GetIndexOfFirstMember``` API should be able trust what it returns

---
In reply to: [426681748](https://github.com/dotnet/roslyn/pull/44231#discussion_r426681748) [](ancestors = 426681748,426204181)